tests/nested/manual/remodel-cross-store/task.yaml

```diff
@@ -0,0 +1,122 @@
+summary: verify a simple UC20 remodel
+details: |
+  Execute a cross-store remodel scenario under the test-snapd brand. The
+  original model uses the same test-snapd-whitebox device view store, while the
+  target model uses test-snapd-remodel store. The new model requires additional
+  snaps that are present in the test-snapd-remodel store.
+
+systems: [ubuntu-18.04-64, ubuntu-20.04-64]
+
+environment:
+  NESTED_CUSTOM_AUTO_IMPORT_ASSERTION: $TESTSLIB/assertions/test-snapd-remodel-auto-import.assert
+  NESTED_IMAGE_ID: uc20-remodel-testing-cross-store
+  NESTED_ENABLE_TPM: false
+  NESTED_ENABLE_SECURE_BOOT: false
+  NESTED_USE_CLOUD_INIT: false
+  SNAPD_TEST_BRAND: test-snapd
+  # store we are remodeling from
+  SNAPD_STORE_WHITEBOX: CQvkdMjgbapk821Po2cF
+  # store we are remodeling to
+  SNAPD_STORE_REMODEL: Ceith6aiSahditeoyizi
+  # device serial, this should match the sub-store mapping rules in the serial
+  # vault. Since this test and remodel-simple are using the same original model,
+  # due to limitations of the snap-store/serial-vault, the serial used in this
+  # test must be different than in remodel-simple.
+  SERIAL: 1234-snapd-remodel-testing-cross-store
+
+prepare: |
+    # shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB"/nested.sh
+    UBUNTU_IMAGE="$GOHOME"/bin/ubuntu-image
+    variant=18
+    if os.query is-focal; then
+        variant=20
+    fi
+    IMAGE_NAME="$(nested_get_image_name core)"
+    # fetch the snapd snap from edge and repack the deb into the snap
+    snap download --channel="latest/edge" snapd
+    "$TESTSTOOLS"/snaps-state repack_snapd_deb_into_snap snapd
+
+    # TODO: once the gadget snap is published, this will download it from the
+    # store
```

Comment:
This is not very clear; did you mean "TODO: once the gadget snap is published, change this code to download it from the store instead"?
Otherwise it seems that the current code will already do the right thing, but then it's not clear why it's a TODO.